contracts/pallet-ibc/src/ics20/mod.rs

```diff
@@ -401,6 +415,55 @@ where
 	}
 }
 
+impl<T> IbcModule<T>
+where
+	T: Config + Send + Sync,
+	u32: From<<T as frame_system::Config>::BlockNumber>,
+	AccountId32: From<<T as frame_system::Config>::AccountId>,
+{
+	fn refund_fee(
+		packet: &Packet,
+		packet_data: &PacketData,
+		callback_name: &str,
+	) -> Result<(), Ics04Error> {
```

Comment:
refund_fee function is called on on_timeout_packet and on_acknowledgement_packet in case of falure when IBC packet did not delivered to destination chain .
this function refund fee that was charged during IBC transfer extrinsic 
and send fee back from FeeAcount to the packet sender account.
will add docs.